File: apps/android/app/src/main/java/com/example/filmfinder/data/remote/model/SearchResponse.kt
package com.example.filmfinder.data.remote.model

/**
 * Data class representing a search response from the OMDb API.
 * Contains a list of search items and metadata.
 */
data class SearchResponse(
    val search: List<SearchItem>,
    val totalResults: String,
    val response: String,
    val page: Int = 1
)

/**
 * Data class representing an individual search result in the search response.
 * Contains basic information about a movie.
 */
data class SearchItem(
    val title: String,
    val year: Int,
    val imdbID: String,
    val type: String,
    val poster: String
)

File: apps/reactNative/src/data/models/SearchResponse.ts
import { Movie, Rating } from "./Movie";

export interface SearchItem {
    title: string;
    year: number;
    imdbID: string;
    type: string;
    poster: string;
}

export interface SearchResponse {
    Search: SearchItem[];
    totalResults: string;
    Response: string;
}

export interface MovieResponse extends Omit<Movie, 'genres' | 'writers' | 'actors' | 'languages' | 'countries'> {
    Genre: string;
    Writer: string;
    Actors: string;
    Language: string;
    Country: string;
    Ratings: Rating[];
}